### PR TITLE
Test fix after runner bump to Java11

### PR DESCRIPTION
**Pull request by Abacn**
Pin PreCommit Java to Java8 for now due to #32208 and #32911

change "failOnError" to false for Javadoc workflow due to #32726. It appears this flag didn't work for Java8 (see #32726)

GCP_IO precommits failed due to 
```
Mockito cannot mock this class: interface org.apache.beam.sdk.io.gcp.firestore.RpcQos$RpcWriteAttempt

Caused by: java.lang.annotation.AnnotationFormatError: Duplicate annotation for class: interface org.checkerframework.checker.nullness.qual.UnknownKeyFor: @org.checkerframework.checker.nullness.qual.UnknownKeyFor()
```
 in Firestore tests.

I think this is a mockito + checkframework + Java11 incompatibility bug, however not able to adjust dependency to resolve it. Removed one of the two generic argument in `RpcWriteAttempt.newFlushBuffer` magically resolved the issue

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
PreCommit Java downgrade Java version not effective due to github action yaml runs on master. Note that intend to bump it back to Java11 after #32912

**Comment by Abacn** [comment]
GCP_IO PreCommit passed: https://github.com/apache/beam/actions/runs/11485499842/job/31965922878?pr=32909

**Review by damccorm — APPROVED** [review]
Thanks

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/RpcQos.java [hunk 1]
```diff
@@ -200,11 +200,10 @@ interface RpcWriteAttempt extends RpcAttempt {
      * provided {@code instant}.
      *
      * @param instant The intended start time of the next rpc
-     * @param <T> The type which will be sent in the request
      * @param <ElementT> The {@link Element} type which the returned buffer will contain
      * @return a new {@link FlushBuffer} which queued messages can be staged to before final flush
      */
-    <T, ElementT extends Element<T>> FlushBuffer<ElementT> newFlushBuffer(Instant instant);
+    <ElementT extends Element<?>> FlushBuffer<ElementT> newFlushBuffer(Instant instant);
```
Comment (kennknowles) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/RpcQos.java:206]: This seems like a regression in functionality. Now the type cannot assure that it is a `FlushBuffer<Element<T>>`

**Comment by kennknowles** [comment]
The Java code that was changed seems like it was more correct and useful. We should perhaps reduce or eliminate use of mockito if it isn't able to cope with pretty basic programming constructs...

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/RpcQos.java (hunk 1 above)
Comment (Abacn) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/RpcQos.java:206]: Admit that it is a compile time regression. Not matter at runtime as generics are erased.

RpcWriteAttempt / RpcWriteAttempt.FlushBuffer / RpcWriteAttempt.Element are package private visible interfaces so user won't feel it

**Comment by Abacn** [comment]
> The Java code that was changed seems like it was more correct and useful. We should perhaps reduce or eliminate use of mockito if it isn't able to cope with pretty basic programming constructs...

Totally agree. Current Firestore tests heavily use mockito, in addition to deeply child test classes which is not recommended

**Pull request merged**